tests/test_models/test_simple_models.py

```diff
@@ -774,3 +794,48 @@ def test_sma_model_predict_components_correct(
 
     target_components_df = forecast.get_target_components()
     np.testing.assert_allclose(target_components_df.values, expected_values)
+
+
+@pytest.mark.parametrize("method", ("predict", "forecast"))
+@pytest.mark.parametrize(
+    "window,seasonality,expected_components_names",
+    (
+        (1, "month", ["target_component_month_lag_1"]),
+        (3, "month", ["target_component_month_lag_1", "target_component_month_lag_2", "target_component_month_lag_3"]),
+        (1, "year", ["target_component_year_lag_1"]),
+    ),
+)
+def test_deadline_ma_predict_components_correct_names(
+    long_periodic_ts, method, window, seasonality, expected_components_names, horizon=10
+):
+    model = DeadlineMovingAverageModel(window=window, seasonality=seasonality)
+    model.fit(ts=long_periodic_ts)
+
+    method_to_call = getattr(model, method)
+    forecast = method_to_call(ts=long_periodic_ts, prediction_size=horizon, return_components=True)
+
+    assert sorted(forecast.target_components_names) == sorted(expected_components_names)
+
+
+@pytest.mark.parametrize("method", ("predict", "forecast"))
+@pytest.mark.parametrize(
+    "window,seasonality,expected_components_names",
```

Comment:
I guess you don't need this parametrization here

Reply:
All of this parameters are used.

Reply:
Fixed